src/components/NewRequest/NewRequest.tsx

```diff
@@ -0,0 +1,174 @@
+import { FC } from 'react';
+import days, { Dayjs } from 'dayjs';
+import { useLocation } from 'react-router-dom';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import TextField from '@mui/material/TextField';
+import { useNavigate } from 'react-router-dom';
+import {
+  NewRequestFormStyled,
+  Input,
+  Label,
+  ErrorMessage,
+  InputContainer,
+  Button,
+  NewRequestContainer,
+  NotesInput,
+  NotesInputContainer,
+} from './NewRequest.styled';
+import { LOCALE_STORAGE_KEY } from 'constants/localeStorage';
+import { Vacation } from 'interfaces';
+
+const NewRequestSchema = Yup.object().shape({
+  vacationType: Yup.string().required('Required'),
+  startDate: Yup.string().required('Required'),
+  endDate: Yup.string().required('Required'),
+  note: Yup.string(),
+});
+
+interface NewRequestFormValues {
+  vacationType: string;
+  startDate: Dayjs;
+  endDate: Dayjs;
+  note: string;
+}
+
+const NewRequest: FC = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const isEdit = location.pathname.split('/edit/')[1];
+
+  const vacations = JSON.parse(
+    localStorage.getItem(LOCALE_STORAGE_KEY) || '[]'
+  );
+
+  let initialValues: NewRequestFormValues = {
```

Comment:
try to avid doing some thing like that. Define a getter function that will provide proper init value

Reply:
done